pkg/daemon/update.go

```diff
@@ -474,6 +597,96 @@ func calculatePostConfigChangeActionFromFileDiffs(diffFileSet []string) (actions
 	return
 }
 
+// calculatePostConfigChangeNodeDisruptionActionFromMCDiffs takes action based on the cluster's Node disruption policies.
+func calculatePostConfigChangeNodeDisruptionActionFromMCDiffs(diffSSH bool, diffFileSet, diffUnitSet []string, clusterPolicies opv1.NodeDisruptionPolicyClusterStatus) []opv1.NodeDisruptionPolicyStatusAction {
+	actions := []opv1.NodeDisruptionPolicyStatusAction{}
+
+	// Step through all file based policies, and build out the actions object
+	for _, diffPath := range diffFileSet {
+		pathFound := false
+		for _, policyFile := range clusterPolicies.Files {
+			klog.Infof("comparing policy path %s to diff path %s", policyFile.Path, diffPath)
```

Comment:
nit: we can probably merge some of the logging (although I guess since it's during an update, the logs should not happen that often). Up to you

Reply:
Will see if I can tighten these up!